rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderBridgeTxsIndexTests.java

```diff
@@ -0,0 +1,378 @@
+        verify(repository, never()).addStorageBytes(
+            any(),
+            any(),
+            any()
+        );
+        // reset calls counter
+        Mockito.reset(repository);
+
+        // Let's set the sigHash and then check if the hasBtcTxSigHash returns true
+        provider.setBridgeBtcTxSigHash(sigHash);
+        verify(repository, times(1)).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString())
+        );
+        // reset calls counter
+        Mockito.reset(repository);
+
+        // The entry should be found in the pending to save set, not in the index
+        boolean shouldFoundSigHash = provider.hasBridgeBtcTxSigHash(sigHash);
+        Assertions.assertTrue(shouldFoundSigHash);
+        verify(repository, never()).getStorageBytes(
+            any(),
+            any()
+        );
+        Mockito.reset(repository);
+
+        // Let's save into the repository the pending to save sigHash
+        provider.save();
+        verify(repository, times(1)).addStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString()),
+            new byte[]{(byte)1}
+        );
+        verify(repository, never()).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString())
+        );
+        Mockito.reset(repository);
+
+        // Check if sigHash exists when there are entries in the index but not pending to save
+        boolean shouldFoundAgainSigHash = provider.hasBridgeBtcTxSigHash(sigHash);
+        Assertions.assertTrue(shouldFoundAgainSigHash);
+        verify(repository, never()).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString())
+        );
+
+        // Create a new bridge storage provider instance to pretend the node is processing a brand-new transaction
+        provider = new BridgeStorageProvider(
+            repository,
+            PrecompiledContracts.BRIDGE_ADDR,
+            bridgeConstantsRegtest,
+            activations
+        );
+
+        // Let's check nothing is persisted when calling save and there are not pending entries to save
+        provider.save();
+        verify(repository, never()).addStorageBytes(
+            any(),
+            any(),
+            any()
+        );
+        Mockito.reset(repository);
+
+        // Try to set a new sigHash that no exists in memory not in the index.
+        Sha256Hash newSigHash = PegTestUtils.createHash(7);
+        provider.setBridgeBtcTxSigHash(newSigHash);
+
+        // Verify the new sigHash is not persisted into the index when save has not been called.
+        verify(repository, never()).addStorageBytes(
+            any(),
```

Comment:
Is there any way to use the actual type for each of the parameters here? And in the other tests.

Reply:
`any()` is not a type. It refers to the value. It can be any value but it must be a value of the type expected for the method. I'm using any here because I expect this method never(`never()`) to get called, so I don't care about the value passing to the method. I care about this method not being called on this test.

Reply:
Oh sorry, I misunderstood your point. I went to the documentation and now I get you what you mean.

![image](https://github.com/rsksmart/rskj/assets/94651730/b5a88f3a-0701-427b-99e7-03efa65f24b5)

According to the `any()` matches everything, including null and `varargs`. On the other hand `any(Class<T> type)` excludes null. For this case, the one I want to is `any()`  because I can guarantee the method is never called no matter what value is passed.

Reply:
Ok.